### PR TITLE
Performance lint: inefficient while condition

### LINKED ISSUE #63: Add `inefficient_while_comp`
# What it does

Checks for an expression like `while a < 10` which is less efficient than `while a != 10`

# Why is this bad

It's less efficient

# Example

```cairo
fn main() {
    let mut a = 1_u32;
    while a < 10 {
        a += 1;
    }
}
```
consumes 199 steps

Use instead:

```cairo
fn main() {
    let mut a = 1_u32;
    while a != 10 {
        a += 1;
    }
}
```
consumes 172 steps

### PR DESCRIPTION
**Pull request by m-kus**
Resolves #63 
Inspired by https://github.com/keep-starknet-strange/raito/issues/156

Note that we cannot automatically fix it because there are false positive cases:
- The rhs arguments is changed in the loop body
- The lhs argument can "skip" the moment where lhs == rhs (e.g. i += 2)

To be 100% sure we need to do deeper checks.

### CODE REVIEW

File: crates/cairo-lint-core/src/lints/performance.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_semantic::{Arenas, Condition, Expr, ExprWhile};
+
+const INEFFICIENT_WHILE_COMP_MESSAGE: &str =
+    "using \"<\" exit condition is inefficient. Consider switching to \"!=\" instead.";
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/lints/performance.rs:7]:
```suggestion
    "using `<` exit condition is inefficient. Consider switching to `!=` instead.";
```

File: crates/cairo-lint-core/tests/test_files/performance/inefficient_while_comp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//! > while loop exit condition less than
```
Comment (0xLucqs) [inline crates/cairo-lint-core/tests/test_files/performance/inefficient_while_comp:1]: what happens in a more complex case ? for example that

```cairo
fn main() {
    let mut a = 1_u32;
    while a < 10 && true {
        a += 1;
    }
}
```

Comment (m-kus) [reply crates/cairo-lint-core/tests/test_files/performance/inefficient_while_comp:1]: Good point, need to go deeper :)

**Review by 0xLucqs — APPROVED** [review]

**Comment by 0xLucqs** [comment]
you need to rebase sorry :)

**Review by 0xLucqs — DISMISSED** [review]

**Review by 0xLucqs — APPROVED** [review]

**Pull request merged**